extra/src/main/java/aima/extra/probability/ProbabilityNumber.java

```diff
@@ -0,0 +1,40 @@
+package aima.extra.probability;
+
+import java.math.BigInteger;
+
+/**
+ * This interface is implemented by the various ProbabilityNumber
+ * implementations.
+ * 
+ * @author Nagaraj Poti
+ *
+ */
+public interface ProbabilityNumber {
```

Comment:
If you look at the Wiki: https://en.wikipedia.org/wiki/Probability
a 'Probability is quantified as a number between 0 and 1' and I think this is what the interface should represent externally (how implementations internally represent that should be hidden, e.g. logspace). 

This would remove the need for the following methods:
* int signum()
* boolean isPosititive()
* boolean isNegative()
* ProbabilityNumber abs()
* ProbabilityNumber negate()
* ProbabilityNumber invert()

I would add the following methods:
* isOne()
* MathContext getMathContext() - so people can get an idea of the precision information used internally.
* BigDecimal getValue(); that will always return a value between 0 and 1inclusive, which is a standardized representation of the internal implementation (e.g. the logspace implemenation would output the exponentiated value not its internal log value).
* boolean sumsToOne(Iterable<ProbabilityNumber> allProbabilities) - this relates to aima3es need to handle rounding thresholds, see: https://github.com/aimacode/aima-java/blob/AIMA3e/aima-core/src/main/java/aima/core/probability/ProbabilityModel.java.

Minor cleanup:
* On an interface there is no need to declare methods 'public' as this is implicit due to being an interface.

As regards implementations of this interface. For now have them only work with their own types (i.e. don't try to mix Log representations with Double representations for now). i.e. have a method:

private ImplementingType toInternalType(ProbabilityNumber pn) {
}

which in a method you would use like this:

@Override
public ProbabilityNumber add(ProbabilityNumber pnAddend) {
    ImplementingType added = totoInternalType(pnAddend);
    ...
}

Later we can decide if we want to be able to mix different implementation types and this would only require updating the 'toInternalType' method with a common shared.